tests/acceptance/pageObjects/FilesPageElement/filesList.js

```diff
@@ -422,60 +423,80 @@ module.exports = {
 
       return { btnSelector, fileActionsBtnSelector }
     },
     /**
      *
      * @param {string} fileName
+     * @param {string} elementType
      *
      * @returns {string}
      */
-    getFileRowSelectorByFileName: function (fileName) {
-      const parts = path.parse(fileName)
-      // If our file has a extension that starts with space (for eg. "newfile. txt")
-      // Then the whole name comes in the filename part (i.e. The file has no extension in such case)
-      // Since the extension has spaces immediately after '.' we check for spaces after removing the '.' using slice().
-      if (parts.ext && !parts.ext.slice(1).startsWith(' ')) {
-        // keep path of nested folders intact, just remove the extension at the end
-        const filePathWithoutExt = parts.dir ? join(parts.dir, parts.name) : parts.name
-        const element = this.elements.fileRowByNameAndExtension
-        return util.format(
-          element.selector,
-          xpathHelper.buildXpathLiteral(filePathWithoutExt),
-          parts.ext
-        )
+    getFileRowSelectorByFileName: function (fileName, elementType = 'file') {
+      if (elementType === 'file') {
+        const parts = path.parse(fileName)
+        // If our file has a extension that starts with space (for eg. "newfile. txt")
+        // Then the whole name comes in the filename part (i.e. The file has no extension in such case)
+        // Since the extension has spaces immediately after '.' we check for spaces after removing the '.' using slice().
+        if (parts.ext && !parts.ext.slice(1).startsWith(' ')) {
+          // keep path of nested folders intact, just remove the extension at the end
+          const filePathWithoutExt = parts.dir ? join(parts.dir, parts.name) : parts.name
+          const element = this.elements.fileRowByNameAndExtension
+          return util.format(
+            element.selector,
+            xpathHelper.buildXpathLiteral(filePathWithoutExt),
+            parts.ext
+          )
+        }
       }
-
       const element = this.elements.fileRowByName
       return util.format(element.selector, xpathHelper.buildXpathLiteral(fileName))
     },
     /**
      *
      * @param {string} fileName
+     * @param {string} elementType
      */
-    getFileLinkSelectorByFileName: function (fileName) {
-      return this.getFileRowSelectorByFileName(fileName) +
-        this.elements.fileLinkInFileRow.selector
+
+    getFileLinkSelectorByFileName: function (fileName, elementType = 'file') {
+      if (elementType === 'file') {
+        return this.getFileRowSelectorByFileName(fileName) +
+          this.elements.fileLinkInFileRow.selector
+      } else if (elementType === 'folder') {
+        fileName = '"' + fileName + '"'
+        return (util.format(this.elements.fileRowByName.selector, fileName) + this.elements.fileLinkInFileRow.selector)
+      } else {
+        return this
+      }
```

Comment:
that logic can go into `getFileRowSelectorByFileName()`

Reply:
done